backend/Wheelingful.BLL/Services/Generic/CountPaginationPages.cs

```diff
@@ -7,10 +7,18 @@ namespace Wheelingful.BLL.Services.Generic;
 
 public class CountPaginationPages<T>(WheelingfulDbContext dbContext) : ICountPaginationPages<T> where T : class
 {
-    public async Task<int> CountByPageSize(CountPagesRequest request)
+    public async Task<int> CountByPageSize(CountPagesRequest request, Func<T, bool>? filter = null)
     {
-        int totalCount = await dbContext.Set<T>().CountAsync();
+        IQueryable<T> query = dbContext.Set<T>();
+
+        if (filter != null)
+        {
+            query = query.Where(filter).AsQueryable();
```

Comment:
AsQueryable looks redundant.
Where here should already be Queryable

Reply:
For some reason it is not Queryable. Not sure why. Will look into it later.